docs/advanced-scenarios.md

```diff
@@ -205,4 +205,103 @@ Luigi allows you to implement and configure feature toggles. They can be used to
+
+- Add CSS styling to position the view container of Luigi
+Specify the left navigation width and position and style the view container. We additionally add a `.spinnerContainer` which is the view loading indicator backdrop.
+```css
+:root {
+  --luigi__left-sidenav--width: 256px;
+}
+.iframeContainer,
+.spinnerContainer {
+  position: absolute;
+  top: $topNavHeight;
+  left: var(--luigi__left-sidenav--width);
+  bottom: 0;
+  right: 0;
+  width: auto;
+  min-width: auto;
+  min-height: auto;
+  display: block;
+}
+.iframeContainer {
+  overflow: auto;
+  -webkit-overflow-scrolling: touch;
+}
+.iframeContainer iframe {
+  border: none;
+  width: 100%;
+  height: 100%;
+  overflow: hidden;
+  margin-bottom: -5px;
+}
+.spinnerContainer {
+  background: rgba(243, 244, 245, 0.8);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+```
+If you are intend to use the tab navigation, it makes sense to specify also the new top view container positioning `.iframeContainerTabNav`.
+```css
+.iframeContainerTabNav {
+  top: calc(#{$topNavHeight} + #{$topNavHeightTab}); // header navigation height PLUS tab navigation height
+}
+```
+In case you are missing more styles and want to have a look at our Luigi code base, take a look at [App.html:1380](https://github.com/SAP/luigi/blob/master/core/src/App.html#L1380)
+
+- Retrieving Luigi events for navigation updates: 
+Each Luigi router update triggers a rebuild of the navigation elements, which you can receive through `Luigi.navigation().addEventListener('topNav')` or its complementary values `leftNav`, `tabNav`, also further events like `routeChanged` or `userInfo` can be received. Show the full list of [navigation events](luigi-core-api.md#addeventlistener).
+
+Example: 
+```javascript
+const id = Luigi.navigation().addEventListener('topNav', data => setTopNavData(data));
+window.onbeforeunload = () => Luigi.navigation().removeEventListener(id); // apply also on component destroy
+```
+
+- Build your navigation based on the data received through the events
+Based on the `data` received above, use `data.topNavData` to build the navigation items, apply category logic, or anything else you want to see implemented. To utilize navigation, use a function that handles navigation clicks. See our [Header.js:38 and :66](https://github.com/SAP/luigi/tree/master/core/examples/luigi-core-material-ui/src/components/Header.js#L38)
+
+Example:
+```javascript
+// template
+<a class="nav-link" onClick="navigateTo(node.pathSegment)">{node.label}</a>
+
+// navigation component
+function navigateTo(pathSegment) {
+  // for top nav
+  Luigi.navigation().navigate(`/${pathSegment}`);
+  // possible usecase on side nav, if each top nav node with children has a `navigationContext` defined
+  Luigi.navigation().fromClosestContext().navigate(pathSegment);
+}
+```
```

Comment:
``````suggestion

- Build your navigation based on the data received through the events.
Based on the `data` received above, use `data.topNavData` to build the navigation items, apply category logic, or anything else you want to implement. To utilize navigation, use a function that handles navigation clicks. See our [Header.js:38 and :66](https://github.com/SAP/luigi/tree/master/core/examples/luigi-core-material-ui/src/components/Header.js#L38)

  - Example:
    ```javascript
    // template
    <a class="nav-link" onClick="navigateTo(node.pathSegment)">{node.label}</a>

    // navigation component
    function navigateTo(pathSegment) {
    // for top nav
    Luigi.navigation().navigate(`/${pathSegment}`);
    // possible usecase on side nav, if each top nav node with children has a `navigationContext` defined
    Luigi.navigation().fromClosestContext().navigate(pathSegment);
    }
    ```
``````